cluster-commands.go

```diff
@@ -0,0 +1,266 @@
+	if err != nil {
+		return err
+	}
+
+	reqData := requestData{
+		relPath: adminAPIPrefix + "/cluster-replicate/join",
+		content: encBuf,
+	}
+
+	resp, err := adm.executeMethod(ctx, http.MethodPut, reqData)
+	defer closeResponse(resp)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return httpRespToErrorResponse(resp)
+	}
+
+	return nil
+}
+
+// BktOp represents the bucket operation being requested.
+type BktOp string
+
+// BktOp value constants.
+const (
+	// make bucket and enable versioning
+	MakeWithVerBktOp BktOp = "make-with-ver"
+	// add replication configuration
+	ConfigureReplBktOp BktOp = "configure-repl"
+	// delete bucket (forceDelete = off)
+	DeleteBucketBktOp BktOp = "delete-bucket"
+	// delete bucket (forceDelete = on)
+	ForceDeleteBucketBktOp BktOp = "force-delete-bucket"
+)
+
+// CRInternalBucketOps - tells peers to create bucket and setup replication.
+func (adm *AdminClient) CRInternalBucketOps(ctx context.Context, bucket string, op BktOp) error {
+	v := url.Values{}
+	v.Add("bucket", bucket)
+	v.Add("operation", string(op))
+	reqData := requestData{
+		queryValues: v,
+		relPath:     adminAPIPrefix + "/cluster-replicate/bucket-ops",
+	}
+
+	resp, err := adm.executeMethod(ctx, http.MethodPut, reqData)
+	defer closeResponse(resp)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return httpRespToErrorResponse(resp)
+	}
+
+	return nil
+}
+
+// CRIAMItem.Type constants.
+const (
+	CRIAMItemPolicy        = "policy"
+	CRIAMItemSvcAcc        = "service-account"
+	CRIAMItemPolicyMapping = "policy-mapping"
+)
+
+// CRSvcAccCreate - create operation
+type CRSvcAccCreate struct {
+	Parent        string            `json:"parent"`
+	AccessKey     string            `json:"accessKey"`
+	SecretKey     string            `json:"secretKey"`
+	Groups        []string          `json:"groups"`
+	LDAPUser      string            `json:"ldapUser"`
+	SessionPolicy *iampolicy.Policy `json:"sessionPolicy"`
+}
+
+// CRSvcAccUpdate - update operation
+type CRSvcAccUpdate struct {
+}
+
+// CRSvcAccDelete - delete operation
+type CRSvcAccDelete struct {
+}
+
+// CRSvcAccChange - sum-type to represent an svc account change.
+type CRSvcAccChange struct {
+	Create *CRSvcAccCreate `json:"crSvcAccCreate"`
+	Update *CRSvcAccUpdate `json:"crSvcAccUpdate"`
+	Delete *CRSvcAccDelete `json:"crSvcAccDelete"`
+}
+
+// CRPolicyMapping - represents mapping of a policy to a user or group.
+type CRPolicyMapping struct {
+	UserOrGroup string `json:"userOrGroup"`
+	IsGroup     bool   `json:"isGroup"`
+	Policy      string `json:"policy"`
+}
+
+// CRIAMItem - represents an IAM object that will be copied to a peer.
+type CRIAMItem struct {
+	Type          string            `json:"type"`
+	Name          string            `json:"name"`
+	Policy        *iampolicy.Policy `json:"policy"`
+	PolicyMapping *CRPolicyMapping  `json:"policyMapping"`
+	SvcAccChange  *CRSvcAccChange   `json:"serviceAccountChange"`
```

Comment:
What is this ? Also maybe we can add IAM type (regular user, sts, svcacc), though I know we are going to support only service accounts for now but maybe not in the future

Reply:
`CRIAMItem` is a sort of "sum" type to represent different kinds of changes related to IAM - I am not sure if that's what you're asking. `SvcAccChange` just represents a service account change such as add/update/delete. If we support regular user, sts, etc we can add corresponding pointer type members to this struct.

Reply:
Added some comments in the code, HTH.